src/App.tsx

```diff
@@ -1,7 +1,33 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.recordKey);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', this.recordKey);
+  }
+
+  recordKey = (event: KeyboardEvent) => {
+    this.setState({ pressedKey: event.key });
+  };
+
+  render() {
+    return (
+      <div className="App">
+        {this.state.pressedKey
+          ? (
+            <p className="App__message">{`The last pressed key is [${this.state.pressedKey}]`}</p>
+          )
+          : (
+            <p className="App__message">Nothing was pressed yet</p>
+          )}
+      </div>
```

Comment:
you can also simplify this code. Since our paragraphs are the same, we can render inside it.
```suggestion
    const { pressedKey } = this.state;

    return (
      <div className="App">
        <p className="App__message">
          {pressedKey
            ? `The last pressed key is [${pressedKey}]`
            : 'Nothing was pressed yet'}
        </p>
      </div>
    );
```